klayout/drc/rule_decks/main.drc

```diff
@@ -14,880 +14,766 @@
+logger.info(format('Wedge enabled  %s', WEDGE))
 
 # BALL
-if $ball == "false"
-  BALL = $ball
-else
-  BALL = "true"
-end # BALL
+BALL = if $ball == 'false'
+         $ball
+       else
+         'true'
+       end
 
-logger.info("Ball enabled  %s" % [BALL])
+logger.info(format('Ball enabled  %s', BALL))
 
 # GOLD
-if $gold == "false"
-  GOLD = $gold
-else
-  GOLD = "true"
-end # GOLD
+GOLD = if $gold == 'false'
+         $gold
+       else
+         'true'
+       end
 
-logger.info("Gold enabled  %s" % [GOLD])
+logger.info(format('Gold enabled  %s', GOLD))
 
-if $mim_option
-  MIM_OPTION = $mim_option
-else
-  MIM_OPTION = "B"
-end
+MIM_OPTION = $mim_option || 'B'
 
-logger.info("MIM Option selected %s" % [MIM_OPTION])
+logger.info(format('MIM Option selected %s', MIM_OPTION))
 
 # OFFGRID
-if $offgrid == "false"
-  OFFGRID = false
-else
-  OFFGRID = true
-end # OFFGRID
+OFFGRID = $offgrid != 'false'
 
-logger.info("Offgrid enabled  %s" % [OFFGRID])
+logger.info(format('Offgrid enabled  %s', OFFGRID))
 
 if $thr
-    threads($thr)
+  threads($thr)
 else
-    thr ||=  Etc.nprocessors
-    threads(thr)
+  thr ||= Etc.nprocessors
+  threads(thr)
 end
-logger.info("Number of threads to use %s" % [$thr])
+logger.info(format('Number of threads to use %s', $thr))
 
 #=== PRINT DETAILS ===
 logger.info("Verbose mode: #{$verbose}")
-if $verbose == "true"
+if $verbose == 'true'
   verbose(true)
 else
   verbose(false)
 end
 
 # === TILING MODE ===
-if $run_mode == "tiling"
+case $run_mode
+when 'tiling'
   tiles(500.um)
   tile_borders(10.um)
-  logger.info("Tiling  mode is enabled.")
+  logger.info('Tiling  mode is enabled.')
 
-elsif $run_mode == "deep"
+when 'deep'
   #=== HIER MODE ===
   deep
-  logger.info("deep  mode is enabled.")
+  logger.info('deep  mode is enabled.')
 else
   #=== FLAT MODE ===
   flat
-  logger.info("flat  mode is enabled.")
-end # run_mode
+  logger.info('flat  mode is enabled.')
+end
 
 # METAL_TOP
-if $metal_top
-  METAL_TOP = $metal_top
-else
-  METAL_TOP = "9K"
-end # METAL_TOP
+METAL_TOP = $metal_top || '9K'
 
-logger.info("METAL_TOP Selected is %s" % [METAL_TOP])
+logger.info(format('METAL_TOP Selected is %s', METAL_TOP))
 
 # METAL_LEVEL
-if $metal_level
-  METAL_LEVEL = $metal_level
-else
-  METAL_LEVEL = "5LM"
-end # METAL_LEVEL
+METAL_LEVEL = $metal_level || '5LM'
 
-logger.info("METAL_STACK Selected is %s" % [METAL_LEVEL])
+logger.info(format('METAL_STACK Selected is %s', METAL_LEVEL))
 
 # FEOL
-if $feol == "false"
+if $feol == 'false'
   FEOL = $feol
-  logger.info("FEOL is disabled.")
+  logger.info('FEOL is disabled.')
 else
-  FEOL = "true"
-  logger.info("FEOL is enabled.")
-end # FEOL
+  FEOL = 'true'.freeze
+  logger.info('FEOL is enabled.')
+end
 
 # BEOL
-if $beol == "false"
+if $beol == 'false'
   BEOL = $beol
-  logger.info("BEOL is disabled.")
+  logger.info('BEOL is disabled.')
 else
-  BEOL = "true"
-  logger.info("BEOL is enabled.")
-end # BEOL
+  BEOL = 'true'.freeze
```

Comment:
same q here?

Reply:
Removed

Reply:
Same as above

Reply:
Can we keep the original structure of the code then and keep any config refactoring for #37?

Reply:
`BEOL` switch is a similar to `FEOL`.

![image](https://user-images.githubusercontent.com/91015308/217466056-077a4803-7f26-4bf7-b867-103f70831fe3.png)